### PR TITLE
[JBPM-9247] Fields attribute isn't processed in Accept header

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

### CODE REVIEW

**Comment by sutaakar** [comment]
@fjtirado I have moved your test to JbpmRestIntegrationTest as it gathers also other header related tests.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java [hunk 1]
```diff
@@ -218,6 +229,21 @@ public void testBasicJbpmRequestManyAcceptHeaders() throws Exception {
...
+
+        long processId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK);
+        ProcessInstance desc = processClient.getProcessInstance(CONTAINER_ID, processId);
+        TaskSummary[] tasks = desc.getActiveUserTasks().getTasks();
+        assertTrue(tasks.length > 0);
+
+        WebTarget target = newRequest(RestURI.build(TestConfig.getKieServerHttpUrl(), QUERY_URI + "/" + TASK_GET_URI, Collections.singletonMap(TASK_INSTANCE_ID, tasks[0].getId())));
+        Assertions.assertThat(target.request(MediaType.APPLICATION_JSON_TYPE).get(String.class)).contains("null");
+        Assertions.assertThat(target.request().header("content-type", MediaType.APPLICATION_JSON + ";fields=not_null;strict=true").get(String.class)).doesNotContain("null");
+        Assertions.assertThat(target.request().header("accept", MediaType.APPLICATION_JSON + ";fields=not_null;strict=true").get(String.class)).doesNotContain("null");
```
Comment (fjtirado) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java:244]: Just for completeness, you can add

Assertions.assertThat(target.request().header("accept", MediaType.APPLICATION_JSON + ";fields=not_null").get(String.class)).doesNotContain("null");

Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java:244]: Good point, added.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java [hunk 2]
```diff
@@ -218,6 +229,22 @@ public void testBasicJbpmRequestManyAcceptHeaders() throws Exception {
...
+        Assertions.assertThat(target.request().header("accept", MediaType.APPLICATION_JSON + ";fields=not_null").get(String.class)).doesNotContain("null");
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java:245]: And just for complete completeness, you can add

`Assertions.assertThat(target.request().header("content-type", MediaType.APPLICATION_JSON + ";fields=not_null").get(String.class)).doesNotContain("null");`

Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java:246]: Added

**Comment by MarianMacik** [comment]
Test failure not related.

**Pull request merged**